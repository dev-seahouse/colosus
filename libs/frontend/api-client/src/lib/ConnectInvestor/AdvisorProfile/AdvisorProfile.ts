import ConnectInvestorBaseApi from '../_Base/Base';
import type { ConnectAdvisorDto } from '@bambu/shared';

export interface ConnectInvestorGetAdvisorProfileResponseDto
  extends ConnectAdvisorDto.IConnectAdvisorProfileForAdvisorDto {
  incomeThreshold: number;
  retireeSavingsThreshold: number;
  advisorSubscriptionIds: 'CONNECT' | 'TRANSACT' | null;
  canPerformTransactActions: boolean;
}

export class ConnectInvestorAdvisorProfileApi extends ConnectInvestorBaseApi {
  constructor(private readonly apiPath = '/advisor-profile') {
    super();
  }

  /**
   * Retrieves some user information associated with the advisor.
   * - {@link http://localhost:9000/openapi#/Connect%20Investor/GetAdvisorProfile}.
   */
  public async getAdvisorProfile() {
    return await this.axios.get<ConnectInvestorGetAdvisorProfileResponseDto>(
      this.apiPath
    );
  }
}

export default ConnectInvestorAdvisorProfileApi;
