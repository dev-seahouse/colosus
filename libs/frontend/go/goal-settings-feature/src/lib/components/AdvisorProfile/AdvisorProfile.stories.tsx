import type { Meta } from '@storybook/react';
import { AdvisorProfile } from './AdvisorProfile';

const Story: Meta<typeof AdvisorProfile> = {
  component: AdvisorProfile,
  title: 'goal-settings/components/AdvisorProfile',
};
export default Story;

export const Primary = {
  args: {},
};
