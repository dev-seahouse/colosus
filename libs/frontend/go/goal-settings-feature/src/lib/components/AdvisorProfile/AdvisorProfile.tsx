import {
  useSelectAdvisorBioQuery,
  useSelectAdvisorFirstNameQuery,
} from '@bambu/go-core';
import { Box, Stack, Typography, MuiLink } from '@bambu/react-ui';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';

import HtmlContent from '../HtmlContent/HtmlContent';

export function AdvisorProfile() {
  const { data } = useSelectAdvisorBioQuery();
  const { data: firstName = '[First Name]' } = useSelectAdvisorFirstNameQuery();

  if (!data || data.content === '') {
    return null;
  }
  return (
    <Box px={3} py={4} sx={{ wordWrap: 'break-word' }}>
      <Stack spacing={1}>
        <Typography fontWeight={600}>{`${firstName}'s Profile`}</Typography>
        <HtmlContent content={data.content} />
        {data.fullProfileLink ? (
          <MuiLink
            underline="always"
            href={data.fullProfileLink}
            target="_blank"
            rel="noopener"
          >
            <Box sx={{ display: 'flex' }}>
              View profile
              <ArrowForwardIcon />
            </Box>
          </MuiLink>
        ) : null}
      </Stack>
    </Box>
  );
}

export default AdvisorProfile;
