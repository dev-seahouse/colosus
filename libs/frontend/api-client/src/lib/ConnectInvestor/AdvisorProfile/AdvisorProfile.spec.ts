import { describe } from 'vitest';
import { ConnectInvestorAdvisorProfileApi } from './AdvisorProfile';

describe('AdvisorProfile', () => {
  const connectInvestorAdvisorProfileApi =
    new ConnectInvestorAdvisorProfileApi();

  describe('connectInvestorAdvisorProfileApi', () => {
    it('should return correct response', async () => {
      const res = await connectInvestorAdvisorProfileApi.getAdvisorProfile();
      expect(res.data).toMatchSnapshot();
    });
  });
});
