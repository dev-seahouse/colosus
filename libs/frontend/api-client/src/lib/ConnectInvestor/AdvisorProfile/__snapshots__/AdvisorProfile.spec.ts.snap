// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdvisorProfile > connectInvestorAdvisorProfileApi > should return correct response 1`] = `
{
  "advisorSubscriptionIds": "CONNECT",
  "advisorSubscriptionInPlace": true,
  "canPerformTransactActions": true,
  "contactMeReasonsRichText": "<p><strong>Let's make it happe</strong>n.</p><p></p><p>Hi there, I'm Robert Henderson, CFP.</p><p>Let's work together to bridge the financial gap and achieve your goals.</p><ul><li><p>Offer diversified investment strategies to minimize risk and optimise the return.</p></li><li><p>Extensive tax planning knowledge to minimize your tax liability .</p></li></ul>",
  "firstName": "Matius",
  "incomeThreshold": 100000,
  "jobTitle": "Financial Planner",
  "lastName": "Ben Slayer",
  "profileBioRichText": "<p>As a financial planner, my primary objective is to help my clients achieve clarity and peace of mind in achieving their financial goals. I prioritize investments that focus on generating income, followed by long-term growth, to help you build a solid financial foundation for your future.</p>",
  "retireeSavingsThreshold": 100000,
  "tenantRealm": "matius_at_bambu_co",
}
`;
